dipy/testing/decorators.py

```diff
@@ -103,3 +104,26 @@ def _set_random_number_generator_wrapper(pytestconfig, *args, **kwargs):
         return _set_random_number_generator_wrapper
 
     return _set_random_number_generator
+
+def warn_on_positional_args(valid=""):
```

Comment:
`valid` may be too vague to know what this argument means. If I understood correctly, this references the version in which the deprecation will start, so maybe the name could be something like: `deprecates_from_version`, `invalid_from_version`.

Reply:
The docstring is missing

Reply:
Noted. Thanks.